src/actions.rs

```diff
@@ -86,63 +144,18 @@ impl ActionHandler {
         match result {
             BuildResult::Success(ref x) | BuildResult::Failure(ref x) => {
                 debug!("build - Success");
+
+                self.clear_build_results();
+
                 {
                     let mut results = self.previous_build_results.lock().unwrap();
```

Comment:
Locking in a function generally doesn't work because the function scopes the lock, you end up having to take a closure which makes the code a bit heavy.